### PR TITLE
Implement pre-forking (still beta)

### PR DESCRIPTION
**Pull request by giampaolo**
TODO: add tests, see http://noosrequired.com/DiegoliToPro/tornado/commit/5ef04877085853298ccbb2bac066f0210cc0a18c?view=inline&w=1

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -346,7 +347,44 @@ Multiple process example:
...
+Pre-fork
+^^^^^^^^
+
+It also exists a third option (UNIX only): the pre-fork model.
+Pre-fork means that a certain number of worker processes are ``spawn()``ed
+before starting the server.
+Each worker process will keep using a 1-thread, async concurrency model,
+handling multiple concurrent connections, but the work load is split.
+This way the delay introduced by a blocking function call is amortized and
+divided by the number of workers, and thus also qthe disk I/O latency is
```
Comment (lurch) [inline docs/tutorial.rst]: `qthe` typo

File: docs/tutorial.rst [hunk 2]
```diff
@@ -346,7 +347,44 @@ Multiple process example:
     if __name__ == "__main__":
         main()
 
+.. _pre-fork:
+
+Pre-fork
+^^^^^^^^
+
+It also exists a third option (UNIX only): the pre-fork model.
```
Comment (lurch) [inline docs/tutorial.rst]: "It also exists" -> "There also exists"

File: docs/tutorial.rst [hunk 3]
```diff
@@ -346,7 +347,44 @@ Multiple process example:
...
+Pre-fork means that a certain number of worker processes are ``spawn()``ed
+before starting the server.
+Each worker process will keep using a 1-thread, async concurrency model,
+handling multiple concurrent connections, but the work load is split.
```
Comment (lurch) [inline docs/tutorial.rst]: "work load" -> "workload"

**Comment by giampaolo** [comment]
@lurch : fixed, thanks man.

**Pull request merged**